user-operator/src/main/java/io/strimzi/operator/user/UserOperatorConfig.java

```diff
@@ -47,313 +141,151 @@ public class UserOperatorConfig {
-        int batchMaxBlockSize = getIntProperty(map, STRIMZI_BATCH_MAXIMUM_BLOCK_SIZE, DEFAULT_BATCH_MAXIMUM_BLOCK_SIZE);
-        int batchMaxBlockTime = getIntProperty(map, STRIMZI_BATCH_MAXIMUM_BLOCK_TIME_MS, DEFAULT_BATCH_MAXIMUM_BLOCK_TIME_MS);
-        int userOperationsThreadPoolSize = getIntProperty(map, STRIMZI_USER_OPERATIONS_THREAD_POOL_SIZE, DEFAULT_USER_OPERATIONS_THREAD_POOL_SIZE);
-
-        String kafkaBootstrapServers = DEFAULT_KAFKA_BOOTSTRAP_SERVERS;
-        String kafkaBootstrapServersEnvVar = map.get(UserOperatorConfig.STRIMZI_KAFKA_BOOTSTRAP_SERVERS);
-        if (kafkaBootstrapServersEnvVar != null && !kafkaBootstrapServersEnvVar.isEmpty()) {
-            kafkaBootstrapServers = kafkaBootstrapServersEnvVar;
-        }
-
-        Labels labels;
-        try {
-            labels = Labels.fromString(map.get(STRIMZI_LABELS));
-        } catch (Exception e)   {
-            throw new InvalidConfigurationException("Failed to parse labels from " + STRIMZI_LABELS, e);
-        }
-
-        String caCertSecretName = map.get(UserOperatorConfig.STRIMZI_CA_CERT_SECRET_NAME);
-        if (caCertSecretName == null || caCertSecretName.isEmpty()) {
-            throw new InvalidConfigurationException(UserOperatorConfig.STRIMZI_CA_CERT_SECRET_NAME + " cannot be null");
+    public UserOperatorConfig(Map<String, String> map) {
+        this.map = new HashMap<>(map.size());
+        for (Map.Entry<String, String> entry : map.entrySet()) {
+            final ConfigParameter<?> configValue = CONFIG_VALUES.get(entry.getKey());
+            if (configValue == null) {
+                throw new IllegalArgumentException("Unknown config key " + entry.getKey());
+            }
+            this.map.put(configValue.key, get(map, configValue));
         }
 
-        String caKeySecretName = map.get(UserOperatorConfig.STRIMZI_CA_KEY_SECRET_NAME);
-        if (caKeySecretName == null || caKeySecretName.isEmpty()) {
-            throw new InvalidConfigurationException(UserOperatorConfig.STRIMZI_CA_KEY_SECRET_NAME + " cannot be null");
+        // now add all those config (with default value) that weren't in the given map
+        Map<String, ConfigParameter<?>> x = new HashMap<>(CONFIG_VALUES);
+        x.keySet().removeAll(map.keySet());
+        for (ConfigParameter<?> value : x.values()) {
+            this.map.put(value.key, get(map, value));
         }
 
-        String clusterCaCertSecretName = map.get(UserOperatorConfig.STRIMZI_CLUSTER_CA_CERT_SECRET_NAME);
-
-        String euoKeySecretName = map.get(UserOperatorConfig.STRIMZI_EO_KEY_SECRET_NAME);
-
-        String caNamespace = map.get(UserOperatorConfig.STRIMZI_CA_NAMESPACE);
-        if (caNamespace == null || caNamespace.isEmpty()) {
-            caNamespace = namespace;
+        if (this.map.get(STRIMZI_CA_NAMESPACE) == null || this.map.get(STRIMZI_CA_NAMESPACE).equals("")) {
+            this.map.put(STRIMZI_CA_NAMESPACE, this.map.get(STRIMZI_NAMESPACE));
```

Comment:
Why are we special casing these when `ConfigParameter` already knows about required-ness?

Reply:
This is just to make sure that if `CA_NAMESPACE is null or empty then we should use the `NAMESPACE` value. It is more like a validation that `CA_NAMESPACE` cannot be null.